alf/environments/suite_metadrive.py

```diff
@@ -0,0 +1,201 @@
+        self._last_step_is_done = True
+
+    @property
+    def batched(self):
+        # TODO(breakds): Add support for multiple algorithm controlled agents in
+        # the future. This environment should be batched in that case.
+        return False
+
+    def env_info_spec(self):
+        return self._env_info_spec
+
+    def observation_spec(self):
+        return self._observation_spec
+
+    def action_spec(self):
+        return self._action_spec
+
+    def render(self, mode):
+        return self._env.render(mode='top_down')
+
+    def _acquire_next_frame(self, action):
+        """Returns the TimeStep give the input action.
+        This is the underlying implementation of both _step() and _reset()
+        1. In _step(), normally it just delegates to this method unless a reset
+           needs to be performed.
+        2. In _reset(), it just delegates to this method after the wrapped
+           environment is reset. The action for _reset() is a simple all-zero
+           action, which makes sense for driving environments.
+        """
+        observation, reward, done, info = self._env.step(action)
+
+        if self._image_channel_first:
+            observation = nest.map_structure(lambda x: x.transpose(2, 0, 1),
+                                             observation)
+
+        discount = [0.0 if done else 1.0]
+
+        self._last_step_is_done = done
+
+        return ds.TimeStep(
+            step_type=ds.StepType.LAST if done else ds.StepType.MID,
+            reward=reward,
+            discount=discount,
+            observation=observation,
+            env_id=self._env_id,
+            prev_action=action,
+            env_info={
+                'velocity@step': info['velocity'],
+                'abs_steering@step': abs(info['steering']),
+                'reach_goal': 1.0 if info['arrive_dest'] else 0.0,
+            })
+
+    def _step(self, action) -> ds.TimeStep:
+        if self._last_step_is_done:
+            return self._reset()
+
+        return self._acquire_next_frame(action)
+
+    def _reset(self) -> ds.TimeStep:
+        _ = self._env.reset()
+
+        # Zero action means do nothing in both longitudinal and lateral
+        first_time_step = self._acquire_next_frame(
+            self._action_spec.zeros().cpu().numpy())
+
+        return first_time_step._replace(step_type=ds.StepType.FIRST)
+
+    def seed(self, seed):
+        pass
+
+    def close(self):
+        self._env.close()
+
+
+@alf.configurable
+def load(
+        map_name,
+        env_id: int = 0,
+        batch_size: int = 1,
+        traffic_density: float = 0.1,
+        environment_num: int = 200,
+        start_seed: int = np.random.randint(10000),
+        decision_repeat: int = 5,  # 0.02 * 5 = 0.1 seconds per action
+        map_size: int = 4):
+    """The loader for metadrive environment.
+
+    See https://metadrive-simulator.readthedocs.io/en/latest/config_system.html
+
+    for details about the configurations.
```

Comment:
map_name and batch_size is not used anywhere and need some comment. Also env_id.

Reply:
`map_name` and `env_id` are used at this moment. Removed `batch_size`.